packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -433,10 +445,47 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
     );
 
     _thumbRect = Offset(x, y) & thumbSize;
-    if (radius == null)
-      canvas.drawRect(_thumbRect!, _paintThumb);
-    else
+
+    if (radius != null) {
       canvas.drawRRect(RRect.fromRectAndRadius(_thumbRect!, radius!), _paintThumb);
+      return;
+    }
+
+    if (shape == null) {
+      canvas.drawRect(_thumbRect!, _paintThumb);
+      return;
+    }
+
+    switch(shape.runtimeType){
```

Comment:
Not sure why we're doing case analysis here. The  value of `shape` can be any OutlinedBorder, not just one of the predefined subclasses. Why is `shape!.paint(canvas, _thumbRect!)` insufficient?

Reply:
If `thumbColor` is not specified for RawScrollbar, this works fine without any case analysis. But if it is, `shape!.paint(canvas, _thumbRect!)` will only draw the border of the shape, without filling the shape with `thumbColor`. This happens in all of the predefined subclasses, because the Paint used to draw on the canvas uses `PaintingStyle.stroke` (which will only draw the border, without filling the shape itself).

Reply:
In that case we may want to consider the option to specify fill when shape is set, or perhaps something else. This hard codes knowledge of other widgets into the scrollbar, which we cannot do.

Reply:
If shape and thumbColor are specified then maybe the scrollbar thumb should be a [PhysicalShape](https://api.flutter.dev/flutter/widgets/PhysicalShape-class.html). That should obviate the case analysis and it supports filled shapes.

Reply:
@HansMuller @Piinks Thanks for the reviews! I may have found a solution to this: First, we get the outerPath of the shape and draw it on the canvas using `_paintThumb` -- this will draw the entire shape using `thumbColor`, without drawing the border. Then, we simply call `shape!.paint(canvas, _thumbRect!)`, which will draw the border _over_ our already drawn path. This will eliminate the entire case analysis and work for any OutlinedBorder. What do you think?

Reply:
@HansMuller I've looked into PhysicalShape, and I was able to draw a scrollbar thumb using a simple SizedBox and a ShapeBorderClipper. However, I wasn't able to draw a border around the shape -- it seems passing a BorderSide to `shape` inside ShapeBorderClipper does not make any difference. Any idea how to make this work?

Reply:
I think you need to make sure that the border is painted on top of the fill. Check out the way the Material widget does this, see the _MaterialInterior class in packages/flutter/lib/src/material/material.dart

Reply:
@HansMuller It worked! Do you think we should go with that approach?

Reply:
We can also keep it as is (i.e. let shape be any OutlinedBorder) and make the changes I've proposed above.

Reply:
I think delegating the problem of rendering a filled/bordered shape to PhysicalShape is worthwhile, since that's what PhysicalShape is for.

Reply:
Alright, so let's suppose we have a PhysicalShape ready to use. How do we use it to actually paint the thumb?

Reply:
> Check out the way the Material widget does this, see the _MaterialInterior class in packages/flutter/lib/src/material/material.dart

Reply:
Yes, I've looked into this - I know how to paint the thumb. I've built a few shapes with PhysicalShape under a Scaffold and so on, so I didn't need to worry about painting -- it just happened automatically. What I mean is, given something like this:
```
final PhysicalShape physicalShape = PhysicalShape(clipper: ...., color: ...., child: ....);
```
Now, how can I use that object to actually paint the thumb on the screen?

Reply:
I've added the solution I mentioned above. I thought it was simpler than using a `PhysicalShape`, and it works for any `OutlinedBorder`.